### PR TITLE
Create Jaeger OTEL agent component 

### LINKED ISSUE #2204: Create Jaeger agent OpenTelemetry distribution
Similar to Jaeger OpenTelemetry collector https://github.com/jaegertracing/jaeger/tree/master/cmd/opentelemetry-collector we should create Jaeger agent based on OpenTelemetry collector. 

There will be also ingester and probably all-in-one based on OTEL collector. Question is whether we should create a separate image for each of these components or have a single image that switches configuration based on a property (env var, flag...)

cc) @jaegertracing/jaeger-maintainers

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #2204

This PR adds a new component - Jaeger Agent based on OpenTelemetry.

Notable changes:
* the image name is `jaeger-opentelemetry-agent
* UDP endpoints were removed from Jaeger OTEL collector
* flag `--reporter.grpc.host-port` is used to configure Jaeger exporter endpoint and also fetch endpoint for sampling strategies in Jaeger receiver.
* collector and agent use the same heath check and metrics endpoints
* agent's main is in the collector's directory we need to use the same go module and reuse parts of the codebase.

### CODE REVIEW

**Comment by pavolloffay** [comment]
@rubenvp8510 or/and @objectiser could you please review?

File: Makefile [hunk 1]
```diff
@@ -310,10 +318,12 @@ docker-images-elastic:
 
 .PHONY: docker-images-jaeger-backend
 docker-images-jaeger-backend:
-	for component in agent collector query ingester opentelemetry-collector ; do \
+	for component in agent collector query ingester ; do \
 		docker build -t $(DOCKER_NAMESPACE)/jaeger-$$component:${DOCKER_TAG} cmd/$$component ; \
 		echo "Finished building $$component ==============" ; \
 	done
+	docker build -t $(DOCKER_NAMESPACE)/jaeger-opentelemetry-collector -f ${OTEL_COLLECTOR_DIR}/cmd/collector/Dockerfile cmd/opentelemetry-collector/cmd/collector
```
Comment (rubenvp8510) [inline Makefile:325]: Why not using the ${DOCKER_TAG} here?

File: cmd/opentelemetry-collector/app/defaults/default_config.go [hunk 2]
```diff
@@ -64,29 +73,19 @@ func Config(storageType string, zipkinHostPort string, factories config.Factorie
...
-func createReceivers(zipkinHostPort string, factories config.Factories) configmodels.Receivers {
+func createCollectorReceivers(zipkinHostPort string, factories config.Factories) configmodels.Receivers {
 	jaeger := factories.Receivers["jaeger"].CreateDefaultConfig().(*jaegerreceiver.Config)
 	// TODO load and serve sampling strategies
 	// TODO bind sampling strategies file
 	jaeger.Protocols = map[string]*receiver.SecureReceiverSettings{
 		"grpc": {
 			ReceiverSettings: configmodels.ReceiverSettings{
-				Endpoint: "localhost:14250",
+				Endpoint: gRPCEndpoint,
```
Comment (rubenvp8510) [inline cmd/opentelemetry-collector/app/defaults/default_config.go:83]: I like that you put this as a constants!

File: cmd/opentelemetry-collector/cmd/agent/main.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+	v := viper.New()
+
+	cmpts := defaults.Components(v)
+	cfgFactory := func(otelViper *viper.Viper, f config.Factories) (*configmodels.Config, error) {
+		cfg := defaults.AgentConfig(cmpts)
+
+		var otelCfg *configmodels.Config
+		if len(app.GetOTELConfigFile()) > 0 {
+			var err error
```
Comment (rubenvp8510) [inline cmd/opentelemetry-collector/cmd/agent/main.go:58]: Why this var err is declared here? is not better in this way: `otelCfg, err := service.FileLoaderConfigFactory(otelViper, f) ` ?

Comment (pavolloffay) [reply cmd/opentelemetry-collector/cmd/agent/main.go:58]: The `:=` would create a new variable `otelCfg` in this case I want to assign to the defined variable above.

File: cmd/opentelemetry-collector/cmd/agent/main.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+		var otelCfg *configmodels.Config
+		if len(app.GetOTELConfigFile()) > 0 {
+			c, err := service.FileLoaderConfigFactory(otelViper, f)
+			if err != nil {
+				return nil, err
+			}
+			otelCfg = c
+		}
+		err := defaults.MergeConfigs(cfg, otelCfg)
```
Comment (objectiser) [inline cmd/opentelemetry-collector/cmd/agent/main.go:64]: Shouldn't the merge be inside the above if block? Merge only required if OTel config file provided?

File: cmd/opentelemetry-collector/cmd/agent/main.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+		ApplicationStartInfo: info,
+		Factories:            cmpts,
+		ConfigFactory:        cfgFactory,
+	})
+	handleErr(err)
+
+	cmd := svc.Command()
+	jconfig.AddFlags(v,
+		cmd,
+		jflags.AddConfigFileFlag,
```
Comment (objectiser) [inline cmd/opentelemetry-collector/cmd/agent/main.go:81]: In the collector main it also has `collectorApp.AddFlags` - should the equivalent `agentApp.AddFlags` be here?

File: crossdock/otel-agent-config.yml [hunk 6]
```diff
@@ -1,18 +1,9 @@
...
 exporters:
   jaeger:
-    endpoint: jaeger-collector:14250
     wait_for_ready: true
 
-service:
-  pipelines:
-    traces:
-      exporters: [jaeger]
+# TODO remove after https://github.com/open-telemetry/opentelemetry-collector/pull/910
```
Comment (objectiser) [inline crossdock/otel-agent-config.yml:5]: Seems to have been merged.

Comment (pavolloffay) [reply crossdock/otel-agent-config.yml:5]: I will update this in a follow-up PR

File: cmd/opentelemetry-collector/cmd/agent/main.go (hunk 5 above)
Comment (pavolloffay) [reply cmd/opentelemetry-collector/cmd/agent/main.go:81]: Not needed if there is no flag we want to consume.

This issue https://github.com/jaegertracing/jaeger/issues/2182 will do some hygiene on flags. We can either expose all or just expose what is used by the new component. For less confusion I think we should expose only what is being used.

File: cmd/opentelemetry-collector/cmd/agent/main.go (hunk 4 above)
Comment (pavolloffay) [reply cmd/opentelemetry-collector/cmd/agent/main.go:64]: The function correctly handles merge if the second parameter is nil

**Pull request merged**